src/sentry/api/endpoints/shared_group_details.py

```diff
@@ -42,8 +45,11 @@ def get(
         except Group.DoesNotExist:
             raise ResourceDoesNotExist
 
+        # Checks if the organization_slug (eventually renamed to organization_id_or_slug) matches the group organization's id or slug
         if organization_slug:
-            if organization_slug != group.organization.slug:
+            if organization_slug != group.organization.slug and organization_slug != str(
```

Comment:
should we be using `isnumeric` rather than checking a stringified version of the org id?